File: georocket-cli/src/test/java/io/georocket/util/DurationFormatTest.java
package io.georocket.util;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * Test {@link DurationFormat}
 * @author Michel Kraemer
 */
public class DurationFormatTest {
  /**
   * Test {@link DurationFormat#format(long)}
   */
  @Test
  public void format() {
    assertEquals("0s", DurationFormat.format(0));
    assertEquals("0.001s", DurationFormat.format(1));
    assertEquals("0.010s", DurationFormat.format(10));
    assertEquals("0.123s", DurationFormat.format(123));
    assertEquals("1s", DurationFormat.format(1000));
    assertEquals("1.024s", DurationFormat.format(1024));
    assertEquals("1m", DurationFormat.format(60000));
    assertEquals("1m 1.024s", DurationFormat.format(60000 + 1024));
    assertEquals("1h", DurationFormat.format(60000 * 60));
    assertEquals("1d", DurationFormat.format(60000 * 60 * 24));
    assertEquals("400d", DurationFormat.format(60000 * 60 * 24 * 400L));
    assertEquals("400d 1h 10m 35.464s", DurationFormat.format(60000 * 60 * 24 * 400L + 4235464));
  }

  /**
   * Test {@link DurationFormat#formatShort(long)}
   */
  @Test
  public void formatShort() {
    assertEquals("00:00:00", DurationFormat.formatShort(0));
    assertEquals("00:00:00", DurationFormat.formatShort(1));
    assertEquals("00:00:00", DurationFormat.formatShort(10));
    assertEquals("00:00:00", DurationFormat.formatShort(123));
    assertEquals("00:00:01", DurationFormat.formatShort(1000));
    assertEquals("00:00:01", DurationFormat.formatShort(1024));
    assertEquals("00:01:00", DurationFormat.formatShort(60000));
    assertEquals("00:01:01", DurationFormat.formatShort(60000 + 1024));
    assertEquals("01:00:00", DurationFormat.formatShort(60000 * 60));
    assertEquals("24:00:00", DurationFormat.formatShort(60000 * 60 * 24));
    assertEquals("9600:00:00", DurationFormat.formatShort(60000 * 60 * 24 * 400L));
    assertEquals("9601:10:35", DurationFormat.formatShort(60000 * 60 * 24 * 400L + 4235464));
  }
}

File: src/test/kotlin/io/georocket/util/DurationFormatTest.kt
package io.georocket.util

import org.junit.Assert.assertEquals
import org.junit.Test

/**
 * Test [formatDuration] and [formatDurationShort]
 */
class DurationFormatTest {
  /**
   * Test [formatDuration]
   */
  @Test
  fun format() {
    assertEquals("0s", 0L.formatDuration())
    assertEquals("0.001s", 1L.formatDuration())
    assertEquals("0.010s", 10L.formatDuration())
    assertEquals("0.123s", 123L.formatDuration())
    assertEquals("1s", 1000L.formatDuration())
    assertEquals("1.024s", 1024L.formatDuration())
    assertEquals("1m", 60000L.formatDuration())
    assertEquals("1m 1.024s", (60000 + 1024L).formatDuration())
    assertEquals("1h", (60000 * 60L).formatDuration())
    assertEquals("1d", (60000 * 60 * 24L).formatDuration())
    assertEquals("400d", (60000 * 60 * 24 * 400L).formatDuration())
    assertEquals("400d 1h 10m 35.464s", (60000 * 60 * 24 * 400L + 4235464).formatDuration())
  }

  /**
   * Test [formatDurationShort]
   */
  @Test
  fun formatShort() {
    assertEquals("00:00:00", 0L.formatDurationShort())
    assertEquals("00:00:00", 1L.formatDurationShort())
    assertEquals("00:00:00", 10L.formatDurationShort())
    assertEquals("00:00:00", 123L.formatDurationShort())
    assertEquals("00:00:01", 1000L.formatDurationShort())
    assertEquals("00:00:01", 1024L.formatDurationShort())
    assertEquals("00:01:00", 60000L.formatDurationShort())
    assertEquals("00:01:01", (60000 + 1024L).formatDurationShort())
    assertEquals("01:00:00", (60000 * 60L).formatDurationShort())
    assertEquals("24:00:00", (60000 * 60 * 24L).formatDurationShort())
    assertEquals("9600:00:00", (60000 * 60 * 24 * 400L).formatDurationShort())
    assertEquals("9601:10:35", (60000 * 60 * 24 * 400L + 4235464).formatDurationShort())
  }
}
